File: SoloLearn Codes/Longest Common Substring/LongestCommonSubstring.java
import java.util.*;

public class LongestCommonSubstring {
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] words = sc.nextLine().split(" ");
        String firstWord = words[0];
        String longestCommonSubstring = String.valueOf(firstWord.charAt(0));

        for (int i = 0; i < firstWord.length(); i++) {
            for (int j = i + 2; j < firstWord.length() + 1; j++) {
                String sub = firstWord.substring(i, j);
                if (Arrays.stream(words).filter(w -> w.contains(sub)).count() == words.length)
                    if (sub.length() > longestCommonSubstring.length())
                        longestCommonSubstring = sub;

            }
        }
        System.out.println(longestCommonSubstring);
    }
}

File: SoloLearn Codes/Longest Common Substring/longest_common_substring.py
words = input().split()
first_word = words[0]
longest_common_substring = first_word[0]

for i in range(len(first_word)):
    for j in range(i + 2, len(first_word) + 1):
        if len([0 for w in words if first_word[i:j] in w]) == len(words):
            if len(first_word[i:j]) > len(longest_common_substring):
                longest_common_substring = first_word[i:j]

print(longest_common_substring)
